From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Michael Goffioul <michael.goffioul@lincor.com>
Date: Mon, 26 Apr 2021 09:01:08 -0400
Subject: [PATCH 1/2] Rename libavutil/time.h to libavutil/avtime.h

This avoid conflicts with libc's time.h. Using "-I bionic/libc/include"
is not working anymore, because bionic now exports the same directory
with -isystem, which supersedes the -I version (as stated in GCC doc).
This means the version of time.h in libavutil will now be found first,
via "-I external/ffmpeg/libavutil" (-I dirs are searched first, before
the -isystem ones).
---
 cmdutils_opencl.c                 | 2 +-
 ffmpeg.c                          | 2 +-
 ffplay.c                          | 2 +-
 ffserver.c                        | 2 +-
 libavcodec/dxva2.c                | 2 +-
 libavcodec/mediacodecdec_common.c | 2 +-
 libavcodec/qsvdec.c               | 2 +-
 libavcodec/qsvenc.c               | 2 +-
 libavcodec/tests/dct.c            | 2 +-
 libavcodec/tests/fft.c            | 2 +-
 libavcodec/tests/motion.c         | 2 +-
 libavdevice/alsa_dec.c            | 2 +-
 libavdevice/alsa_enc.c            | 2 +-
 libavdevice/avfoundation.m        | 2 +-
 libavdevice/bktr.c                | 2 +-
 libavdevice/decklink_dec.cpp      | 2 +-
 libavdevice/fbdev_dec.c           | 2 +-
 libavdevice/gdigrab.c             | 2 +-
 libavdevice/jack.c                | 2 +-
 libavdevice/openal-dec.c          | 2 +-
 libavdevice/oss_dec.c             | 2 +-
 libavdevice/pulse_audio_dec.c     | 2 +-
 libavdevice/pulse_audio_enc.c     | 2 +-
 libavdevice/qtkit.m               | 2 +-
 libavdevice/sdl2.c                | 2 +-
 libavdevice/sndio_dec.c           | 2 +-
 libavdevice/v4l2-common.h         | 2 +-
 libavdevice/xcbgrab.c             | 2 +-
 libavfilter/avf_showcqt.c         | 2 +-
 libavfilter/f_bench.c             | 2 +-
 libavfilter/f_realtime.c          | 2 +-
 libavfilter/setpts.c              | 2 +-
 libavfilter/vf_deinterlace_qsv.c  | 2 +-
 libavfilter/vf_scale_qsv.c        | 2 +-
 libavformat/avio.c                | 2 +-
 libavformat/fifo.c                | 2 +-
 libavformat/hls.c                 | 2 +-
 libavformat/hlsproto.c            | 2 +-
 libavformat/http.c                | 2 +-
 libavformat/mux.c                 | 2 +-
 libavformat/network.c             | 2 +-
 libavformat/nutenc.c              | 2 +-
 libavformat/rtmphttp.c            | 2 +-
 libavformat/rtpdec.c              | 2 +-
 libavformat/rtsp.c                | 2 +-
 libavformat/rtspdec.c             | 2 +-
 libavformat/rtspenc.c             | 2 +-
 libavformat/sapenc.c              | 2 +-
 libavformat/segment.c             | 2 +-
 libavformat/tcp.c                 | 2 +-
 libavformat/tests/fifo_muxer.c    | 2 +-
 libavformat/udp.c                 | 2 +-
 libavformat/utils.c               | 2 +-
 libavformat/wavenc.c              | 2 +-
 libavformat/webm_chunk.c          | 2 +-
 libavutil/{time.h => avtime.h}    | 0
 libavutil/eval.c                  | 2 +-
 libavutil/hwcontext_qsv.c         | 2 +-
 libavutil/parseutils.c            | 2 +-
 libavutil/tests/des.c             | 2 +-
 libavutil/time.c                  | 2 +-
 tools/aviocat.c                   | 2 +-
 tools/pktdumper.c                 | 2 +-
 63 files changed, 62 insertions(+), 62 deletions(-)
 rename libavutil/{time.h => avtime.h} (100%)

diff --git a/cmdutils_opencl.c b/cmdutils_opencl.c
index 906aef4836..032347c650 100644
--- a/cmdutils_opencl.c
+++ b/cmdutils_opencl.c
@@ -19,7 +19,7 @@
  */
 
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/log.h"
 #include "libavutil/opencl.h"
 #include "libavutil/avstring.h"
diff --git a/ffmpeg.c b/ffmpeg.c
index cd19594f8b..4bdac8e461 100644
--- a/ffmpeg.c
+++ b/ffmpeg.c
@@ -60,7 +60,7 @@
 #include "libavutil/imgutils.h"
 #include "libavutil/timestamp.h"
 #include "libavutil/bprint.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/threadmessage.h"
 #include "libavcodec/mathops.h"
 #include "libavformat/os_support.h"
diff --git a/ffplay.c b/ffplay.c
index c0b326c8cc..54520e0b09 100644
--- a/ffplay.c
+++ b/ffplay.c
@@ -39,7 +39,7 @@
 #include "libavutil/parseutils.h"
 #include "libavutil/samplefmt.h"
 #include "libavutil/avassert.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavformat/avformat.h"
 #include "libavdevice/avdevice.h"
 #include "libswscale/swscale.h"
diff --git a/ffserver.c b/ffserver.c
index f9f987acac..351704b974 100644
--- a/ffserver.c
+++ b/ffserver.c
@@ -47,7 +47,7 @@
 #include "libavutil/rational.h"
 #include "libavutil/parseutils.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 #include <stdarg.h>
 #if HAVE_UNISTD_H
diff --git a/libavcodec/dxva2.c b/libavcodec/dxva2.c
index 0036b5bec1..5aeedc9759 100644
--- a/libavcodec/dxva2.c
+++ b/libavcodec/dxva2.c
@@ -24,7 +24,7 @@
 #include <string.h>
 
 #include "libavutil/log.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 #include "avcodec.h"
 #include "dxva2_internal.h"
diff --git a/libavcodec/mediacodecdec_common.c b/libavcodec/mediacodecdec_common.c
index 2ec25c581d..5ac609d935 100644
--- a/libavcodec/mediacodecdec_common.c
+++ b/libavcodec/mediacodecdec_common.c
@@ -27,7 +27,7 @@
 #include "libavutil/mem.h"
 #include "libavutil/log.h"
 #include "libavutil/pixfmt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/timestamp.h"
 
 #include "avcodec.h"
diff --git a/libavcodec/qsvdec.c b/libavcodec/qsvdec.c
index d7664ce581..d2dec8c354 100644
--- a/libavcodec/qsvdec.c
+++ b/libavcodec/qsvdec.c
@@ -33,7 +33,7 @@
 #include "libavutil/log.h"
 #include "libavutil/pixdesc.h"
 #include "libavutil/pixfmt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 #include "avcodec.h"
 #include "internal.h"
diff --git a/libavcodec/qsvenc.c b/libavcodec/qsvenc.c
index 57bc83a47f..108faeb22d 100644
--- a/libavcodec/qsvenc.c
+++ b/libavcodec/qsvenc.c
@@ -30,7 +30,7 @@
 #include "libavutil/hwcontext_qsv.h"
 #include "libavutil/mem.h"
 #include "libavutil/log.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/imgutils.h"
 #include "libavcodec/bytestream.h"
 
diff --git a/libavcodec/tests/dct.c b/libavcodec/tests/dct.c
index 29af3fea8a..8b82867882 100644
--- a/libavcodec/tests/dct.c
+++ b/libavcodec/tests/dct.c
@@ -38,7 +38,7 @@
 #include "libavutil/common.h"
 #include "libavutil/internal.h"
 #include "libavutil/lfg.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 #include "libavcodec/dct.h"
 #include "libavcodec/idctdsp.h"
diff --git a/libavcodec/tests/fft.c b/libavcodec/tests/fft.c
index 83f2ff2a08..ebc8c0c297 100644
--- a/libavcodec/tests/fft.c
+++ b/libavcodec/tests/fft.c
@@ -41,7 +41,7 @@
 #include "libavutil/lfg.h"
 #include "libavutil/log.h"
 #include "libavutil/mathematics.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 #if AVFFT
 #include "libavcodec/avfft.h"
diff --git a/libavcodec/tests/motion.c b/libavcodec/tests/motion.c
index d89f9408c2..8b2dd804b8 100644
--- a/libavcodec/tests/motion.c
+++ b/libavcodec/tests/motion.c
@@ -32,7 +32,7 @@
 #include "libavutil/internal.h"
 #include "libavutil/lfg.h"
 #include "libavutil/mem.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 #undef printf
 
diff --git a/libavdevice/alsa_dec.c b/libavdevice/alsa_dec.c
index c50ce71506..bce70113e5 100644
--- a/libavdevice/alsa_dec.c
+++ b/libavdevice/alsa_dec.c
@@ -50,7 +50,7 @@
 #include "libavutil/internal.h"
 #include "libavutil/mathematics.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 #include "libavformat/internal.h"
 
diff --git a/libavdevice/alsa_enc.c b/libavdevice/alsa_enc.c
index 0bef625ea4..7225a4bb0b 100644
--- a/libavdevice/alsa_enc.c
+++ b/libavdevice/alsa_enc.c
@@ -40,7 +40,7 @@
 #include <alsa/asoundlib.h>
 
 #include "libavutil/internal.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 
 #include "libavformat/internal.h"
diff --git a/libavdevice/avfoundation.m b/libavdevice/avfoundation.m
index e2ddf47dbe..d6dae68236 100644
--- a/libavdevice/avfoundation.m
+++ b/libavdevice/avfoundation.m
@@ -34,7 +34,7 @@
 #include "libavformat/internal.h"
 #include "libavutil/internal.h"
 #include "libavutil/parseutils.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "avdevice.h"
 
 static const int avf_time_base = 1000000;
diff --git a/libavdevice/bktr.c b/libavdevice/bktr.c
index 418247dc4e..fefe1f4e0f 100644
--- a/libavdevice/bktr.c
+++ b/libavdevice/bktr.c
@@ -29,7 +29,7 @@
 #include "libavutil/log.h"
 #include "libavutil/opt.h"
 #include "libavutil/parseutils.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #if HAVE_DEV_BKTR_IOCTL_METEOR_H && HAVE_DEV_BKTR_IOCTL_BT848_H
 # include <dev/bktr/ioctl_meteor.h>
 # include <dev/bktr/ioctl_bt848.h>
diff --git a/libavdevice/decklink_dec.cpp b/libavdevice/decklink_dec.cpp
index 39974e3ff4..c7dd329d93 100644
--- a/libavdevice/decklink_dec.cpp
+++ b/libavdevice/decklink_dec.cpp
@@ -33,7 +33,7 @@ extern "C" {
 #include "libavutil/avutil.h"
 #include "libavutil/common.h"
 #include "libavutil/imgutils.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/mathematics.h"
 #if CONFIG_LIBZVBI
 #include <libzvbi.h>
diff --git a/libavdevice/fbdev_dec.c b/libavdevice/fbdev_dec.c
index 3b31373fd6..f850a374b9 100644
--- a/libavdevice/fbdev_dec.c
+++ b/libavdevice/fbdev_dec.c
@@ -38,7 +38,7 @@
 #include "libavutil/log.h"
 #include "libavutil/mem.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/parseutils.h"
 #include "libavutil/pixdesc.h"
 #include "libavformat/internal.h"
diff --git a/libavdevice/gdigrab.c b/libavdevice/gdigrab.c
index 87f5012034..63827c067f 100644
--- a/libavdevice/gdigrab.c
+++ b/libavdevice/gdigrab.c
@@ -31,7 +31,7 @@
 #include "config.h"
 #include "libavformat/internal.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include <windows.h>
 
 /**
diff --git a/libavdevice/jack.c b/libavdevice/jack.c
index 076078ce6d..f90671e4d1 100644
--- a/libavdevice/jack.c
+++ b/libavdevice/jack.c
@@ -28,7 +28,7 @@
 #include "libavutil/log.h"
 #include "libavutil/fifo.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavcodec/avcodec.h"
 #include "libavformat/avformat.h"
 #include "libavformat/internal.h"
diff --git a/libavdevice/openal-dec.c b/libavdevice/openal-dec.c
index 6eb0efe38f..e16563c7b8 100644
--- a/libavdevice/openal-dec.c
+++ b/libavdevice/openal-dec.c
@@ -25,7 +25,7 @@
 #include <AL/alc.h>
 
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavformat/internal.h"
 #include "avdevice.h"
 
diff --git a/libavdevice/oss_dec.c b/libavdevice/oss_dec.c
index 9f748f2bc3..b595101bf5 100644
--- a/libavdevice/oss_dec.c
+++ b/libavdevice/oss_dec.c
@@ -37,7 +37,7 @@
 
 #include "libavutil/internal.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 #include "libavcodec/avcodec.h"
 
diff --git a/libavdevice/pulse_audio_dec.c b/libavdevice/pulse_audio_dec.c
index 95a1d6ecfa..2cb9f709eb 100644
--- a/libavdevice/pulse_audio_dec.c
+++ b/libavdevice/pulse_audio_dec.c
@@ -26,7 +26,7 @@
 
 #include "libavutil/internal.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 #include "libavformat/avformat.h"
 #include "libavformat/internal.h"
diff --git a/libavdevice/pulse_audio_enc.c b/libavdevice/pulse_audio_enc.c
index 0efcf0fe7e..d8126ac7b1 100644
--- a/libavdevice/pulse_audio_enc.c
+++ b/libavdevice/pulse_audio_enc.c
@@ -25,7 +25,7 @@
 #include "libavformat/internal.h"
 #include "libavutil/internal.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/log.h"
 #include "libavutil/attributes.h"
 #include "pulse_audio_common.h"
diff --git a/libavdevice/qtkit.m b/libavdevice/qtkit.m
index 22a94ca561..accb6928ed 100644
--- a/libavdevice/qtkit.m
+++ b/libavdevice/qtkit.m
@@ -36,7 +36,7 @@
 #include "libavutil/opt.h"
 #include "libavformat/internal.h"
 #include "libavutil/internal.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "avdevice.h"
 
 #define QTKIT_TIMEBASE 100
diff --git a/libavdevice/sdl2.c b/libavdevice/sdl2.c
index 5d9e91ec21..5a47058d7f 100644
--- a/libavdevice/sdl2.c
+++ b/libavdevice/sdl2.c
@@ -31,7 +31,7 @@
 #include "libavutil/opt.h"
 #include "libavutil/parseutils.h"
 #include "libavutil/pixdesc.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "avdevice.h"
 
 typedef struct {
diff --git a/libavdevice/sndio_dec.c b/libavdevice/sndio_dec.c
index 2d13232bf1..fc1f9bedfd 100644
--- a/libavdevice/sndio_dec.c
+++ b/libavdevice/sndio_dec.c
@@ -24,7 +24,7 @@
 
 #include "libavutil/internal.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 #include "libavformat/avformat.h"
 #include "libavformat/internal.h"
diff --git a/libavdevice/v4l2-common.h b/libavdevice/v4l2-common.h
index 0cbaec8bb9..fc98c1f0c4 100644
--- a/libavdevice/v4l2-common.h
+++ b/libavdevice/v4l2-common.h
@@ -43,7 +43,7 @@
 #include "timefilter.h"
 #include "libavutil/parseutils.h"
 #include "libavutil/pixdesc.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/avstring.h"
 
 struct fmt_map {
diff --git a/libavdevice/xcbgrab.c b/libavdevice/xcbgrab.c
index 1968fe02d4..d113f7a26c 100644
--- a/libavdevice/xcbgrab.c
+++ b/libavdevice/xcbgrab.c
@@ -41,7 +41,7 @@
 #include "libavutil/mathematics.h"
 #include "libavutil/opt.h"
 #include "libavutil/parseutils.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 #include "libavformat/avformat.h"
 #include "libavformat/internal.h"
diff --git a/libavfilter/avf_showcqt.c b/libavfilter/avf_showcqt.c
index 875ba48cee..7de9286ad4 100644
--- a/libavfilter/avf_showcqt.c
+++ b/libavfilter/avf_showcqt.c
@@ -25,7 +25,7 @@
 #include "libavutil/xga_font_data.h"
 #include "libavutil/eval.h"
 #include "libavutil/pixdesc.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "avfilter.h"
 #include "internal.h"
 #include "lavfutils.h"
diff --git a/libavfilter/f_bench.c b/libavfilter/f_bench.c
index f9e1006925..a5e06ba459 100644
--- a/libavfilter/f_bench.c
+++ b/libavfilter/f_bench.c
@@ -17,7 +17,7 @@
  */
 
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "avfilter.h"
 #include "formats.h"
 #include "internal.h"
diff --git a/libavfilter/f_realtime.c b/libavfilter/f_realtime.c
index 171c16aaaa..67c4e6b50a 100644
--- a/libavfilter/f_realtime.c
+++ b/libavfilter/f_realtime.c
@@ -19,7 +19,7 @@
  */
 
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "avfilter.h"
 #include "internal.h"
 
diff --git a/libavfilter/setpts.c b/libavfilter/setpts.c
index 4505498bf3..a341b0697e 100644
--- a/libavfilter/setpts.c
+++ b/libavfilter/setpts.c
@@ -30,7 +30,7 @@
 #include "libavutil/internal.h"
 #include "libavutil/mathematics.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "audio.h"
 #include "avfilter.h"
 #include "internal.h"
diff --git a/libavfilter/vf_deinterlace_qsv.c b/libavfilter/vf_deinterlace_qsv.c
index 2810bffe46..539b07338c 100644
--- a/libavfilter/vf_deinterlace_qsv.c
+++ b/libavfilter/vf_deinterlace_qsv.c
@@ -34,7 +34,7 @@
 #include "libavutil/mathematics.h"
 #include "libavutil/opt.h"
 #include "libavutil/pixdesc.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 #include "avfilter.h"
 #include "formats.h"
diff --git a/libavfilter/vf_scale_qsv.c b/libavfilter/vf_scale_qsv.c
index a5f5be7d66..ac85cbb3e4 100644
--- a/libavfilter/vf_scale_qsv.c
+++ b/libavfilter/vf_scale_qsv.c
@@ -35,7 +35,7 @@
 #include "libavutil/mathematics.h"
 #include "libavutil/opt.h"
 #include "libavutil/pixdesc.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 #include "avfilter.h"
 #include "formats.h"
diff --git a/libavformat/avio.c b/libavformat/avio.c
index 1e79c9dd5c..55fbea5892 100644
--- a/libavformat/avio.c
+++ b/libavformat/avio.c
@@ -22,7 +22,7 @@
 #include "libavutil/avstring.h"
 #include "libavutil/dict.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/avassert.h"
 #include "os_support.h"
 #include "avformat.h"
diff --git a/libavformat/fifo.c b/libavformat/fifo.c
index 2cbe5c56af..c8664f6479 100644
--- a/libavformat/fifo.c
+++ b/libavformat/fifo.c
@@ -21,7 +21,7 @@
 
 #include "libavutil/avassert.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/thread.h"
 #include "libavutil/threadmessage.h"
 #include "avformat.h"
diff --git a/libavformat/hls.c b/libavformat/hls.c
index 01731bd36b..88a4bf39d4 100644
--- a/libavformat/hls.c
+++ b/libavformat/hls.c
@@ -32,7 +32,7 @@
 #include "libavutil/mathematics.h"
 #include "libavutil/opt.h"
 #include "libavutil/dict.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "avformat.h"
 #include "internal.h"
 #include "avio_internal.h"
diff --git a/libavformat/hlsproto.c b/libavformat/hlsproto.c
index 2b19ed0cf6..784d26ca1e 100644
--- a/libavformat/hlsproto.c
+++ b/libavformat/hlsproto.c
@@ -26,7 +26,7 @@
  */
 
 #include "libavutil/avstring.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "avformat.h"
 #include "avio_internal.h"
 #include "internal.h"
diff --git a/libavformat/http.c b/libavformat/http.c
index d06103ab6d..0604b43fd4 100644
--- a/libavformat/http.c
+++ b/libavformat/http.c
@@ -28,7 +28,7 @@
 #include "libavutil/avassert.h"
 #include "libavutil/avstring.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/parseutils.h"
 
 #include "avformat.h"
diff --git a/libavformat/mux.c b/libavformat/mux.c
index e1e49a81be..4d46456b50 100644
--- a/libavformat/mux.c
+++ b/libavformat/mux.c
@@ -35,7 +35,7 @@
 #include "libavutil/internal.h"
 #include "libavutil/mathematics.h"
 #include "libavutil/parseutils.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "riff.h"
 #include "audiointerleave.h"
 #include "url.h"
diff --git a/libavformat/network.c b/libavformat/network.c
index b3987a4d11..c78a5c908c 100644
--- a/libavformat/network.c
+++ b/libavformat/network.c
@@ -25,7 +25,7 @@
 #include "libavcodec/internal.h"
 #include "libavutil/avutil.h"
 #include "libavutil/mem.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 int ff_tls_init(void)
 {
diff --git a/libavformat/nutenc.c b/libavformat/nutenc.c
index a92ff55c01..e127ed4b42 100644
--- a/libavformat/nutenc.c
+++ b/libavformat/nutenc.c
@@ -26,7 +26,7 @@
 #include "libavutil/tree.h"
 #include "libavutil/dict.h"
 #include "libavutil/avassert.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/opt.h"
 #include "libavcodec/bytestream.h"
 #include "libavcodec/mpegaudiodata.h"
diff --git a/libavformat/rtmphttp.c b/libavformat/rtmphttp.c
index ef6146ca86..3d043c8bc4 100644
--- a/libavformat/rtmphttp.c
+++ b/libavformat/rtmphttp.c
@@ -27,7 +27,7 @@
 #include "libavutil/avstring.h"
 #include "libavutil/intfloat.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "internal.h"
 #include "http.h"
 #include "rtmp.h"
diff --git a/libavformat/rtpdec.c b/libavformat/rtpdec.c
index 4acb1ca629..5a060d2918 100644
--- a/libavformat/rtpdec.c
+++ b/libavformat/rtpdec.c
@@ -22,7 +22,7 @@
 #include "libavutil/mathematics.h"
 #include "libavutil/avstring.h"
 #include "libavutil/intreadwrite.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 #include "avformat.h"
 #include "network.h"
diff --git a/libavformat/rtsp.c b/libavformat/rtsp.c
index 261e970b75..711181cd17 100644
--- a/libavformat/rtsp.c
+++ b/libavformat/rtsp.c
@@ -28,7 +28,7 @@
 #include "libavutil/random_seed.h"
 #include "libavutil/dict.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "avformat.h"
 #include "avio_internal.h"
 
diff --git a/libavformat/rtspdec.c b/libavformat/rtspdec.c
index 0cb8f53a4c..0069cbb182 100644
--- a/libavformat/rtspdec.c
+++ b/libavformat/rtspdec.c
@@ -23,7 +23,7 @@
 #include "libavutil/intreadwrite.h"
 #include "libavutil/mathematics.h"
 #include "libavutil/random_seed.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "avformat.h"
 
 #include "internal.h"
diff --git a/libavformat/rtspenc.c b/libavformat/rtspenc.c
index e7707bb5fb..c6f0ccddc0 100644
--- a/libavformat/rtspenc.c
+++ b/libavformat/rtspenc.c
@@ -31,7 +31,7 @@
 #include "avio_internal.h"
 #include "libavutil/intreadwrite.h"
 #include "libavutil/avstring.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "url.h"
 
 #define SDP_MAX_SIZE 16384
diff --git a/libavformat/sapenc.c b/libavformat/sapenc.c
index 3098e340ca..5b56870251 100644
--- a/libavformat/sapenc.c
+++ b/libavformat/sapenc.c
@@ -25,7 +25,7 @@
 #include "libavutil/avstring.h"
 #include "libavutil/dict.h"
 #include "libavutil/intreadwrite.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "internal.h"
 #include "network.h"
 #include "os_support.h"
diff --git a/libavformat/segment.c b/libavformat/segment.c
index 8ec3653b38..9deb706a99 100644
--- a/libavformat/segment.c
+++ b/libavformat/segment.c
@@ -40,7 +40,7 @@
 #include "libavutil/avstring.h"
 #include "libavutil/parseutils.h"
 #include "libavutil/mathematics.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/timecode.h"
 #include "libavutil/time_internal.h"
 #include "libavutil/timestamp.h"
diff --git a/libavformat/tcp.c b/libavformat/tcp.c
index 07b4ed9fa3..c4852cbf6e 100644
--- a/libavformat/tcp.c
+++ b/libavformat/tcp.c
@@ -22,7 +22,7 @@
 #include "libavutil/avassert.h"
 #include "libavutil/parseutils.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 #include "internal.h"
 #include "network.h"
diff --git a/libavformat/tests/fifo_muxer.c b/libavformat/tests/fifo_muxer.c
index e20bd6e7b2..783d81eb32 100644
--- a/libavformat/tests/fifo_muxer.c
+++ b/libavformat/tests/fifo_muxer.c
@@ -21,7 +21,7 @@
 
 #include <stdlib.h>
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/avassert.h"
 #include "libavformat/avformat.h"
 #include "libavformat/url.h"
diff --git a/libavformat/udp.c b/libavformat/udp.c
index 3835f989c4..855551ae66 100644
--- a/libavformat/udp.c
+++ b/libavformat/udp.c
@@ -36,7 +36,7 @@
 #include "libavutil/avstring.h"
 #include "libavutil/opt.h"
 #include "libavutil/log.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "internal.h"
 #include "network.h"
 #include "os_support.h"
diff --git a/libavformat/utils.c b/libavformat/utils.c
index 38d247c6cd..c8c290aadf 100644
--- a/libavformat/utils.c
+++ b/libavformat/utils.c
@@ -32,7 +32,7 @@
 #include "libavutil/opt.h"
 #include "libavutil/parseutils.h"
 #include "libavutil/pixdesc.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/time_internal.h"
 #include "libavutil/timestamp.h"
 
diff --git a/libavformat/wavenc.c b/libavformat/wavenc.c
index 7f3059e99a..cf496f56c9 100644
--- a/libavformat/wavenc.c
+++ b/libavformat/wavenc.c
@@ -37,7 +37,7 @@
 #include "libavutil/intreadwrite.h"
 #include "libavutil/mathematics.h"
 #include "libavutil/opt.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/time_internal.h"
 
 #include "avformat.h"
diff --git a/libavformat/webm_chunk.c b/libavformat/webm_chunk.c
index f8dbaa3339..d199c67040 100644
--- a/libavformat/webm_chunk.c
+++ b/libavformat/webm_chunk.c
@@ -38,7 +38,7 @@
 #include "libavutil/avstring.h"
 #include "libavutil/parseutils.h"
 #include "libavutil/mathematics.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavutil/time_internal.h"
 #include "libavutil/timestamp.h"
 
diff --git a/libavutil/time.h b/libavutil/avtime.h
similarity index 100%
rename from libavutil/time.h
rename to libavutil/avtime.h
diff --git a/libavutil/eval.c b/libavutil/eval.c
index 638259adef..274a2d449f 100644
--- a/libavutil/eval.c
+++ b/libavutil/eval.c
@@ -35,7 +35,7 @@
 #include "internal.h"
 #include "log.h"
 #include "mathematics.h"
-#include "time.h"
+#include "avtime.h"
 #include "avstring.h"
 #include "timer.h"
 #include "reverse.h"
diff --git a/libavutil/hwcontext_qsv.c b/libavutil/hwcontext_qsv.c
index 5550ffe143..c6acf3c08f 100644
--- a/libavutil/hwcontext_qsv.c
+++ b/libavutil/hwcontext_qsv.c
@@ -38,7 +38,7 @@
 #include "mem.h"
 #include "pixfmt.h"
 #include "pixdesc.h"
-#include "time.h"
+#include "avtime.h"
 
 typedef struct QSVDevicePriv {
     AVBufferRef *child_device_ctx;
diff --git a/libavutil/parseutils.c b/libavutil/parseutils.c
index 7ca07b37a1..599df44fc8 100644
--- a/libavutil/parseutils.c
+++ b/libavutil/parseutils.c
@@ -31,7 +31,7 @@
 #include "random_seed.h"
 #include "time_internal.h"
 #include "parseutils.h"
-#include "time.h"
+#include "avtime.h"
 
 #ifdef TEST
 
diff --git a/libavutil/tests/des.c b/libavutil/tests/des.c
index 309be29473..1c7df4f061 100644
--- a/libavutil/tests/des.c
+++ b/libavutil/tests/des.c
@@ -23,7 +23,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 
 static uint64_t rand64(void)
 {
diff --git a/libavutil/time.c b/libavutil/time.c
index afa6658aa6..10c4e6f671 100644
--- a/libavutil/time.c
+++ b/libavutil/time.c
@@ -33,7 +33,7 @@
 #include <windows.h>
 #endif
 
-#include "time.h"
+#include "avtime.h"
 #include "error.h"
 
 int64_t av_gettime(void)
diff --git a/tools/aviocat.c b/tools/aviocat.c
index 983108a64a..8878f72faa 100644
--- a/tools/aviocat.c
+++ b/tools/aviocat.c
@@ -21,7 +21,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavformat/avformat.h"
 
 static int usage(const char *argv0, int ret)
diff --git a/tools/pktdumper.c b/tools/pktdumper.c
index 6516ad3a22..f0ebba68e8 100644
--- a/tools/pktdumper.c
+++ b/tools/pktdumper.c
@@ -32,7 +32,7 @@
 #endif
 
 #include "libavutil/avstring.h"
-#include "libavutil/time.h"
+#include "libavutil/avtime.h"
 #include "libavformat/avformat.h"
 
 #define FILENAME_BUF_SIZE 4096
